tools/scylla-nodetool.cc

```diff
@@ -20,6 +26,45 @@ const auto app_name = "scylla-nodetool";
 logging::logger nodetool_log(app_name);
 
 class scylla_rest_client {
+    sstring _host;
+    http::experimental::client _api_client;
+
+    static socket_address resolve_to_socket_address(const sstring& host, uint16_t port) {
```

Comment:
You might want to re-use the `s3::dns_connection_factory` thing, it also "supports" https

Reply:
I will look into it, best be forward compatible. The REST API doesn't support https at the moment, nor any authentication/authorization, which is the main reason we always recommend binding it to `127.0.0.1`. Which is a shame, because nodetool (both old and new) is capable of connection to remote nodes. The java nodetool even has some auth stuff, which is of course completely defeated because our JMX connects to scylla via plain HTTP.

Reply:
We now have TLS client certificate authentication, it should be easy to give access only to the superuser.

But let's focus on the minimal changes now to support the current configuration.

Remote nodetool can be supported by tunneling a local port to the server via ssh.